File: app/src/main/java/ru/iandreyshev/parserrss/ui/fragment/AddRssDialog.java
package ru.iandreyshev.parserrss.ui.fragment;

import android.app.Dialog;
import android.support.v4.app.FragmentManager;
import android.content.Context;
import android.content.DialogInterface;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v7.app.AlertDialog;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.EditText;

import com.arellomobile.mvp.MvpAppCompatDialogFragment;

import ru.iandreyshev.parserrss.R;
import ru.iandreyshev.parserrss.ui.activity.FeedActivity;
import ru.iandreyshev.parserrss.ui.listeners.IOnSubmitAddRssListener;

public class AddRssDialog extends MvpAppCompatDialogFragment {
    private IOnSubmitAddRssListener mOnSubmitListener;
    private EditText mField;

    public static void show(final FragmentManager fragmentManager) {
        new AddRssDialog().show(fragmentManager, AddRssDialog.class.getName());
    }

    @Override
    @NonNull
    public Dialog onCreateDialog(final Bundle savedInstanceState) {
        final View view = LayoutInflater.from(getContext()).inflate(R.layout.add_feed_dialog, null);
        mField = view.findViewById(R.id.add_feed_dialog_url_field);

        return new AlertDialog.Builder(view.getContext())
                .setView(view)
                .setPositiveButton(R.string.add_feed_dialog_submit_button, this::onAddButtonClick)
                .setNegativeButton(R.string.add_feed_dialog_cancel_button, null)
                .create();
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        mOnSubmitListener = (FeedActivity) context;
    }

    private void onAddButtonClick(DialogInterface dialog, int which) {
        if (mOnSubmitListener != null) {
            mOnSubmitListener.onSubmitAddRss(mField.getText().toString());
        }
    }
}

File: app/src/main/kotlin/ru/iandreyshev/parserrss/ui/fragment/AddRssDialog.kt
package ru.iandreyshev.parserrss.ui.fragment

import android.app.Dialog
import android.content.Context
import android.os.Bundle
import android.support.v7.app.AlertDialog
import android.text.InputFilter
import android.view.LayoutInflater
import android.widget.TextView

import com.arellomobile.mvp.MvpAppCompatDialogFragment

import ru.iandreyshev.parserrss.R

import kotlinx.android.synthetic.main.dialog_add_feed.*
import kotlinx.android.synthetic.main.dialog_add_feed.view.*
import java.lang.ref.WeakReference

class AddRssDialog : MvpAppCompatDialogFragment() {

    companion object {
        private const val MAX_FIELD_LENGTH = 1024

        fun newInstance(url: String): AddRssDialog {
            val dialog = AddRssDialog()
            dialog.mUrl = url

            return dialog
        }
    }

    interface IListener {
        fun onAddRss(url: String)
    }

    private var mListener: WeakReference<IListener>? = null
    private var mUrl: String? = null

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val view = LayoutInflater.from(context).inflate(R.layout.dialog_add_feed, null)

        view.urlField.filters = Array(1, { _ -> InputFilter.LengthFilter(MAX_FIELD_LENGTH) })
        view.urlField.setText(mUrl ?: "", TextView.BufferType.EDITABLE)

        return AlertDialog.Builder(view.context)
                .setTitle(R.string.add_feed_dialog_title)
                .setView(view)
                .setPositiveButton(R.string.add_feed_dialog_submit_button, { _, _ -> mListener?.get()?.onAddRss(dialog.urlField.text.toString()) })
                .setNegativeButton(R.string.add_feed_dialog_cancel_button, null)
                .create()
    }

    override fun onAttach(context: Context?) {
        super.onAttach(context)

        if (context is IListener) {
            mListener = WeakReference(context)
        }
    }
}
